### ISSUE TITLE
Default DateBoundary in adding encounter-based datasets should be "Anytime"

### ISSUE THREAD

**Issue by jnothman** [uwrit/leaf#489]
While I appreciate that in many instances the user will not want to capture "anytime", it is very easy for the user to miss the date boundary panel if they are not seeking to filter on dates, and very likely that a user wanting to filter on dates will look directly at the same panel.

**Comment by ndobb** [comment]
I tend to agree - in fact, in early versions of Leaf the default was indeed "Anytime". The issue we ran into was that for potentially very large datasets (e.g., all blood pressure measurements ever for a large cohort), queries would often timeout and leave users confused / annoyed. The imperfect compromise we settled on was to default on "Past 6 Months", which I agree is easy to miss, but at least queries tend to be less heavy on the DB.

I'm okay with potentially reverting back to "Anytime" as default (as incidentally at UW we have a much more performant prod DB than in the past).
